service/implementation/auto_request_api/sport_data_managers/teams_data_manager.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional
+
+from database.models import Sport
+from database.session import SessionLocal
+from dto.api_input import TeamsLeagueDTO
+from service.api_logic.teams_logic import get_teams
+from service.implementation.auto_request_api.sport_data_managers.abstract_sport_data_manager import \
+    AbstractSportDataManager
+from service.implementation.auto_request_api.sport_data_managers.sport_consts import get_team_statistics_url, \
+    get_team_statistics_index, get_host, get_team_index
+
+
+class TeamsDataManager(AbstractSportDataManager):
+    _teams__sport_id: Optional[int]
+    _leagues__api_id: Optional[int]
+    _countries__api_id: Optional[int]
+    _page: Optional[int]
+    _per_page: Optional[int]
+
+    def __init__(self, data_manager):
```

Comment:
would be awesome if we could do something like: `self, data_manager: SomeSpecificDataManagerInterface`

just to make it more developer-friendly

Reply:
I presume it is a dictionary here. If that is true then "data_manager" is not the best variable name for it (since it's just storeing data, not managing it)